core/src/test/java/org/springframework/security/authentication/UserDetailsRepositoryReactiveAuthenticationManagerTests.java

```diff
@@ -171,4 +172,90 @@ public void authenticateWhenPostAuthenticationChecksNotSet() {
 
 		verifyZeroInteractions(this.postAuthenticationChecks);
 	}
+
+	@Test
+	public void testAuthenticateFailsIfAccountExpired() {
+		this.manager.setPasswordEncoder(this.encoder);
+		this.manager.setUserDetailsService(new MockUserDetailsServiceUserAccountExpired());
```

Comment:
Could we use `this.userDetailsService` here instead? It would be better if this test had the same pattern as the other tests by using `when(this.userDetailsService.findByUsername(any())).thenReturn(Mono.just(expiredUser));`